server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -161,6 +164,13 @@
 
   private volatile Throwable persistError;
 
+  private final boolean isRealTime;
+  // Use next Map to store metadata (File, SegmentId) for a hydrant for batch appenderator
+  // in order to facilitate the mapping of the QueryableIndex associated with a given hydrant
+  // at merge time. This is necessary since batch appenderator will not map the QueryableIndex
+  // at persist time in order to minimize its memory footprint.
+  private final Map<FireHydrant, Pair<File, SegmentId>> persistedHydrantMetadata = new HashMap<>();
```

Comment:
does `FireHydrant` has a `hashCode` implementation? Maybe using identityHashMap makes more sense here? Also, do we ever need to clear this map? like when `org.apache.druid.segment.realtime.appenderator.Appenderator#clear` or `org.apache.druid.segment.realtime.appenderator.Appenderator#drop` is called.

Reply:
Yeah, IndentyHashMap makes it explicit that the key is a Java reference. Also, clearing the Map in the places you referenced above.